src/Data/NormalKey.purs

```diff
@@ -82,26 +80,37 @@ toApple key = key
 combinationToApple :: Set NormalKey -> Set NormalKey
 combinationToApple = toList >>> map toApple >>> fromList
 
-instance showKey :: Show Key where
-  show (Key s) = "(Key " ++ show s ++ ")"
+data NormalKeyView = Modifier Int | Other String | Enter
+
+toView :: NormalKey -> NormalKeyView
+toView (NormalKey "Control") = Modifier 0
+toView (NormalKey "Meta") = Modifier 1
+toView (NormalKey "Alt") = Modifier 2
+toView (NormalKey "Shift") = Modifier 3
+toView (NormalKey "Enter") = Enter
+toView (NormalKey s) = Other s
 
 instance eqNormalKey :: Eq NormalKey where
   eq (NormalKey x) (NormalKey y) = eq x y
 
+instance eqNormalKeyView :: Eq NormalKeyView where
+  eq (Modifier x) (Modifier y) = eq x y
+  eq (Other x) (Other y) = eq x y
+  eq Enter Enter = true
+  eq _ _ = false
+
+instance ordNormalKeyView :: Ord NormalKeyView where
+  compare (Modifier x) (Modifier y) = compare x y
+  compare (Modifier _) _ = GT
+  compare _ (Modifier _) = LT
+  compare (Other x) (Other y) = compare x y
+  compare Enter Enter = EQ
+  compare Enter _ = LT
+  compare _ Enter = GT
```

Comment:
@beckyconning I think it looks like a valid instance now. But if you said you wanted `Enter` to come after all the other keys, wouldn't you need to reverse this clause?